src/app/content/components/Toolbar/PracticeQuestionsButton.tsx

```diff
@@ -26,15 +29,20 @@ const PracticeQuestionsText = styled.span`
 // tslint:disable-next-line:variable-name
 const PracticeQuestionsButton = () => {
   const dispatch = useDispatch();
+  const {book, page} = useSelector(bookAndPage);
   const isEnabled = useSelector(practiceQuestionsEnabled);
   const trackOpenClose = useAnalyticsEvent('openClosePracticeQuestions');
   const hasPracticeQs = useSelector(hasPracticeQuestions);
 
   if (!isEnabled || !hasPracticeQs) { return null; }
 
   const openPracticeQuestionsSummary = () => {
-    dispatch(openPracticeQuestions());
-    trackOpenClose();
+    const section = book && page ? findArchiveTreeNodeById(book.tree, page.id) : null;
+    if (section && isLinkedArchiveTreeSection(section)) {
+      dispatch(setSelectedSection(section));
+      dispatch(openPracticeQuestions());
```

Comment:
we would still need `setSelectedSection` for chapter filtering and we also do not pass anything for actions that are opening other modals
I think it doesn't really matter a lot because we will soon implement opening modals on location change anyway

Reply:
> I think it doesn't really matter a lot because we will soon implement opening modals on location change anyway

is a good argument for going back to the hook you deleted, if the modal was opened by a location change we'd need to select the default section somehow